lib/contract/api/api.go

```diff
@@ -1,15 +1,69 @@
 package api
 
 import (
-	"boscoin.io/sebak/lib/contract/payload"
-	"boscoin.io/sebak/lib/contract/storage"
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/contract/context"
 	"boscoin.io/sebak/lib/contract/value"
+	"time"
 )
 
-type API interface {
-	Helloworld(string) (string, error)
-	GetStorageItem(string) (*storage.StorageItem, error)
-	PutStorageItem(string, *storage.StorageItem) error
-	CallContract(*payload.ExecCode) (*value.Value, error)
-	GetBlockHeight() int64
+//TODO: For prototype testing
+func (a *API) Helloworld(greeting string) (string, error) {
+	return greeting + " WORLD!!", nil
+}
+
+// Contract API is used to executors
+type API struct {
+	contractAddress string // the current contract address
+	ctx             *context.Context
+}
+
+func NewAPI(ctx *context.Context, contractAddr string) *API {
+	api := &API{
+		contractAddress: contractAddr,
+		ctx:             ctx,
+	}
+	return api
+}
+
+// Read a item of a key from this contract own storage
+func (a *API) GetStorageItem(key []byte) (item *value.Value, err error) {
+	hashKey := sebakcommon.BytesToHash(key)
+	encoded := a.ctx.StateDB().GetState(a.contractAddress, hashKey)
+	item = new(value.Value)
+	item.Deserialize(encoded)
+
+	return item, nil
+}
+
+// Write a item to this contract's own storage
+func (a *API) PutStorageItem(key []byte, value *value.Value) (err error) {
+	hashKey := sebakcommon.BytesToHash(key)
+	encoded := value.Serialize()
+	a.ctx.StateDB().SetState(a.contractAddress, hashKey, encoded)
+	return
+}
+
+// Get this contract's balance
+func (a *API) GetBalance() (balance *value.Value, err error) {
+	balanceString := a.ctx.StateDB().GetBalance(a.contractAddress)
+	amount := sebakcommon.MustAmountFromString(balanceString)
+	balance, err = value.ToValue(uint64(amount))
+	return balance, nil
+}
+
+// Call another contract
```

Comment:
Why do you comment on this method? :-)

Reply:
I am not sure that we need call another contract from a contract.
now, it may cause circular import. 
I will handle it when we need that feature.